### PR TITLE
Poetry Environment

### PR DESCRIPTION
**Pull request by ntlhui**
Adds Poetry for package/environment management

### CODE REVIEW

**Comment by ccrutchf** [comment]
One thing to consider is how well does `poetry.lock` handle cross-architecutre.  Did not used to be well, but looks like that may have changed since I've given it a try.

https://github.com/python-poetry/poetry/issues/5205

File: PyHa/birdnet_lite/analyze.py [hunk 1]
```diff
@@ -83,13 +83,14 @@ def splitSignal(sig, rate, overlap, seconds=3.0, minlen=1.5):
 def readAudioData(path, overlap, sample_rate=48000):
 
     print('READING AUDIO DATA...', end=' ', flush=True)
-
-    # Open file with librosa (uses ffmpeg or libav)
+    print("Path: ", path)
```
Comment (ntlhui) [inline PyHa/birdnet_lite/analyze.py:86]: Please use python format strings (https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals)

File: PyHa/birdnet_lite/analyze.py [hunk 2]
```diff
@@ -83,13 +83,14 @@ def splitSignal(sig, rate, overlap, seconds=3.0, minlen=1.5):
...
+    # Open file with librosa (uses ffmanaeg or libav)
     try:
         sig, rate = librosa.load(path, sr=sample_rate, mono=True, res_type='kaiser_fast')
         clip_length = librosa.get_duration(y=sig, sr=rate)
-    except:
-        return 0
+    except Exception as e:
+        print(e)
```
Comment (ntlhui) [inline PyHa/birdnet_lite/analyze.py:92]: E4E pylint disallows `e` as a variable name.  Use `exp` or `exc`.
E4E pylint discourages broad-except.  Please reconsider catching `builtins.Exception`

File: PyHa/microfaune_package/microfaune/audio.py [hunk 3]
```diff
@@ -124,7 +124,7 @@ def create_spec(data, fs, n_mels=32, n_fft=2048, hop_len=1024):
     """
     # Calculate spectrogram
     S = librosa.feature.melspectrogram(
-      data, sr=fs, n_fft=n_fft, hop_length=hop_len, n_mels=n_mels)
+    y=data, sr=fs, n_fft=n_fft, hop_length=hop_len, n_mels=n_mels)
```
Comment (ntlhui) [inline PyHa/microfaune_package/microfaune/audio.py:127]: Recommend each kwarg on a separate line

File: PyHa/statistics.py [hunk 4]
```diff
@@ -263,25 +263,25 @@ def automated_labeling_statistics(
...
             elif stats_type == "IoU":
                 IoU_Matrix = clip_IoU(clip_automated_df, clip_manual_df)
                 clip_stats_df = matrix_IoU_Scores(
                     IoU_Matrix, clip_manual_df, threshold)
                 if statistics_df.empty:
                     statistics_df = clip_stats_df
                 else:
-                    statistics_df = statistics_df.append(clip_stats_df)
+                    statistics_df = pd.concat([statistics_df, clip_stats_df])
         except BaseException as e:
```
Comment (ntlhui) [inline PyHa/statistics.py:275]: E4E pylint discourages broad-except.  Please consider not catching `builtins.BaseException`
E4E pylint does not allow `e` as a variable name.  Please consider `exc` or `exp`

File: PyHa/tweetynet_package/tweetynet/Load_data_functions.py [hunk 5]
```diff
@@ -216,12 +216,12 @@ def predictions_to_kaleidoscope(predictions, SIGNAL, audio_dir, audio_file, manu
         raise BaseException("No birds were detected!!")
 
     if offset.iloc[0] != 0:
-        kaleidoscope_df.append(pd.DataFrame({"OFFSET": [0], "DURATION": [offset.iloc[0]]}))
-    kaleidoscope_df.append(intermediary_df[intermediary_df["DURATION"] >= 2*time_bin_seconds])
+        pd.concat([kaleidoscope_df,pd.DataFrame({"OFFSET": [0], "DURATION": [offset.iloc[0]]})])
```
Comment (ntlhui) [inline PyHa/tweetynet_package/tweetynet/Load_data_functions.py:219]: `kaleidoscope_df` is defined on line 213 as a list!

Comment (ntlhui) [reply PyHa/tweetynet_package/tweetynet/Load_data_functions.py]: Rolled back in 46fa363

**Review by Sean1572 — APPROVED** [review]
Probably should put some of the raised concerns into github issues before submitting this PR. I do agree with changing the error handling but that I think can safely be its own PR. 

Other than that code changes lgtm

**Pull request merged**